### PR TITLE
Rholang: Normalizer: Ensure match target is propagated

### PR DESCRIPTION
**Pull request by KentShikama**
This is follow up from the same issue with if/else (see 41ca611).

### CODE REVIEW

File: rholang/src/test/scala/coop/rchain/rholang/interpreter/NormalizeTest.scala [hunk 1]
```diff
@@ -543,6 +543,39 @@ class ProcMatcherSpec extends FlatSpec with Matchers {
...
+    val listBindings = new ListName()
+    listBindings.add(new NameQuote(pMatch))
+    val listLinearBinds = new ListLinearBind()
+    listLinearBinds.add(new LinearBindImpl(listBindings, new NameQuote(new PNil())))
+    val linearSimple = new LinearSimple(listLinearBinds)
+    val receipt      = new ReceiptLinear(linearSimple)
+    var input        = new PInput(receipt, new PNil())
+
+    val result    = ProcNormalizeMatcher.normalizeMatch(input, inputs)
+    val bindCount = 2
```
Comment (KentShikama) [inline rholang/src/test/scala/coop/rchain/rholang/interpreter/NormalizeTest.scala]: Just wanted to put particular attention to this bindCount. I guess the alternative that is possible is 0.

File: rholang/src/test/scala/coop/rchain/rholang/interpreter/NormalizeTest.scala [hunk 2]
```diff
@@ -543,6 +543,39 @@ class ProcMatcherSpec extends FlatSpec with Matchers {
       ProcNormalizeMatcher.normalizeMatch(pMatch, inputs)
     }
   }
+  "PMatch" should "Handle a match inside a for pattern" in {
+    // for (@(match (x | y) { case 47 => Nil }) <- @Nil) { Nil }
```
Comment (guardbotmk3) [inline rholang/src/test/scala/coop/rchain/rholang/interpreter/NormalizeTest.scala]: Can you write the example according to the grammar?
match {x | y} {
  47 => Nil
}

File: rholang/src/test/scala/coop/rchain/rholang/interpreter/NormalizeTest.scala (hunk 1 above)
Comment (guardbotmk3) [reply rholang/src/test/scala/coop/rchain/rholang/interpreter/NormalizeTest.scala]: Nope, it should be two. The match target has 2 free variables, and so the whole match expression does as well.

Comment (KentShikama) [reply rholang/src/test/scala/coop/rchain/rholang/interpreter/NormalizeTest.scala]: Ok good to confirm.

File: rholang/src/test/scala/coop/rchain/rholang/interpreter/NormalizeTest.scala (hunk 2 above)
Comment (KentShikama) [reply rholang/src/test/scala/coop/rchain/rholang/interpreter/NormalizeTest.scala]: Done

**Pull request merged**